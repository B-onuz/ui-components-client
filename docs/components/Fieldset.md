# Fieldset (Component)

Componente de formulário.

## Como utilizar

```js
import Fieldset from 'ui-components/lib/components/Fieldset'
```

```jsx
import Fieldset from 'ui-components/lib/components/Fieldset'
import TextField from 'ui-components/lib/components/TextField'

const MyForm = () => {
  return (
    <Fieldset mt={4} legend={'Informações pessoais'}>
      <TextField id="name" name={'name'} required label={'Nome'} m={2} placeholder={'Nome'} />
    </Fieldset>
  )
}
```

## Styled-system props

- space https://styled-system.com/table#space
- layout https://styled-system.com/table#layout
