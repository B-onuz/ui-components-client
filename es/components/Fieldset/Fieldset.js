"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _Legend = _interopRequireDefault(require("../Legend"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Fieldset/Fieldset.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  padding: 0;\n  margin: 0;\n  ", "\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledFieldset = _styledComponents["default"].fieldset(_templateObject(), function (_ref) {
  var bordered = _ref.bordered;
  return !bordered ? "\n    border: none;\n  " : "\n    border-radius: 4px;\n    border: 1px solid #DEDEDF;\n    legend {\n      margin: 1em;\n      padding: .3em;\n    }\n  ";
}, (0, _styledSystem.compose)(_styledSystem.space, _styledSystem.layout));

var Fieldset = function Fieldset(_ref2) {
  var children = _ref2.children,
      legend = _ref2.legend,
      rest = _objectWithoutPropertiesLoose(_ref2, ["children", "legend"]);

  return /*#__PURE__*/_react["default"].createElement(StyledFieldset, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 27,
      columnNumber: 5
    }
  }), !!legend && /*#__PURE__*/_react["default"].createElement(_Legend["default"], {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 28,
      columnNumber: 20
    }
  }, legend), children);
};

var _default = Fieldset;
exports["default"] = _default;