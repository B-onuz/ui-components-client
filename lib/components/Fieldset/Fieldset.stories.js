'use strict'

exports.__esModule = true
exports.FieldsetExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Fieldset = _interopRequireDefault(require('./Fieldset'))

var _TextField = _interopRequireDefault(require('../TextField'))

var _reflexbox = require('reflexbox')

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Fieldset/Fieldset.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Fieldset',
  component: _Fieldset['default'],
}
exports['default'] = _default

var FieldsetExample = function FieldsetExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _react['default'].Fragment,
    null,
    /*#__PURE__*/ _react['default'].createElement(
      _Fieldset['default'],
      {
        m: 4,
        legend: 'Informações pessoais',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 13,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _reflexbox.Flex,
        {
          flexWrap: 'wrap',
          m: -2,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 14,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Box,
          {
            width: [1, 1 / 2, 1 / 3],
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 15,
              columnNumber: 9,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
            id: 'name',
            name: 'name',
            required: true,
            label: 'Nome',
            placeholder: 'Nome',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 16,
              columnNumber: 11,
            },
          })
        ),
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Box,
          {
            width: [1, 1 / 2, 1 / 3],
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 18,
              columnNumber: 9,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
            id: 'lastname',
            name: 'lastname',
            label: 'Sobrenome',
            placeholder: 'Sobrenome',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 19,
              columnNumber: 11,
            },
          })
        ),
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Box,
          {
            width: [1, 1 / 2, 1 / 3],
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 21,
              columnNumber: 9,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
            id: 'fullname',
            name: 'fullname',
            label: 'Exibir Como:',
            placeholder: 'Nome + Sobrenome',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 22,
              columnNumber: 11,
            },
          })
        )
      )
    ),
    /*#__PURE__*/ _react['default'].createElement(
      _Fieldset['default'],
      {
        m: 4,
        bordered: true,
        p: '1em',
        legend: 'Dados de contato (bordered)',
        __self: _this,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 26,
          columnNumber: 5,
        },
      },
      /*#__PURE__*/ _react['default'].createElement(
        _reflexbox.Flex,
        {
          flexWrap: 'wrap',
          m: -2,
          __self: _this,
          __source: {
            fileName: _jsxFileName,
            lineNumber: 27,
            columnNumber: 7,
          },
        },
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Box,
          {
            width: [1, 1 / 2, 1 / 3],
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 28,
              columnNumber: 9,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
            id: 'email',
            type: 'email',
            name: 'email',
            label: 'Email',
            placeholder: 'Email',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 29,
              columnNumber: 11,
            },
          })
        ),
        /*#__PURE__*/ _react['default'].createElement(
          _reflexbox.Box,
          {
            width: [1, 1 / 2, 1 / 3],
            m: 2,
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 31,
              columnNumber: 9,
            },
          },
          /*#__PURE__*/ _react['default'].createElement(_TextField['default'], {
            id: 'telefone',
            type: 'phone',
            name: 'fullname',
            label: 'Telefone',
            placeholder: 'Telefone',
            __self: _this,
            __source: {
              fileName: _jsxFileName,
              lineNumber: 32,
              columnNumber: 11,
            },
          })
        )
      )
    )
  )
}

exports.FieldsetExample = FieldsetExample
