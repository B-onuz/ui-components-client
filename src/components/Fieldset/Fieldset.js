import React from 'react'
import styled from 'styled-components'
import { layout, space, compose } from 'styled-system'
import Legend from '../Legend'

const StyledFieldset = styled.fieldset`
  padding: 0;
  margin: 0;
  ${({ bordered }) =>
    !bordered
      ? `
    border: none;
  `
      : `
    border-radius: 4px;
    border: 1px solid #DEDEDF;
    legend {
      margin: 1em;
      padding: .3em;
    }
  `}
  ${compose(space, layout)}
`

const Fieldset = ({ children, legend, ...rest }) => {
  return (
    <StyledFieldset {...rest}>
      {!!legend && <Legend>{legend}</Legend>}
      {children}
    </StyledFieldset>
  )
}

export default Fieldset
