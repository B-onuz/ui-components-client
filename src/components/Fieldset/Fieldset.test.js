import React from 'react'
import Fieldset from './Fieldset.js'
import renderer from 'react-test-renderer'

test('Fieldset changes the class when hovered', () => {
  const component = renderer.create(<Fieldset />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
