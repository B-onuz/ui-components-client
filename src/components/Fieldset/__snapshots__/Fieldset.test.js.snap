// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fieldset changes the class when hovered 1`] = `
<fieldset
  className="sc-AxirZ hglgNQ"
/>
`;
